github/messages.go

```diff
@@ -132,17 +138,40 @@ func messageMAC(signature string) ([]byte, func() hash.Hash, error) {
 //       // Process payload...
 //     }
 //
-func ValidatePayload(r *http.Request, secretKey []byte) (payload []byte, err error) {
-	payload, err = ioutil.ReadAll(r.Body)
-	if err != nil {
-		return nil, err
+func ValidatePayload(r *http.Request, secretKey []byte) ([]byte, error) {
+	// jsonPayload is the json webhook data
+	var jsonPayload []byte
+	// rawPayload is the raw body that GitHub uses to calculate the signature
+	var rawPayload []byte
```

Comment:
Style wise, it might be nicer to do this:

```Go
var (
    jsonPayload []byte // JSON webhook data.
    rawPayload  []byte // Raw body that GitHub uses to calculate the signature.
)
```

But it's up to you, use what you think looks nicer.